File: binary-search/binary-search.js
let search = function(nums, target) {
    let start = 0
    let end = nums.length

    console.log(nums.length)

    while (start <= end) {
        let middle = Math.floor((start + end) / 2)
        let current = nums[middle]
        if (current === target) {
            return middle
        }

        if (target > current) {
            start = middle + 1
        } else if (target < current) {
            end = middle - 1
        }
    }

    return -1
};

File: binary-search/binary-search.go
package main

import (
	"math"
)

func search(nums []int, target int) int {
	start := 0
	end := len(nums) - 1

	for start <= end {
		middle := int(math.Floor(float64(start+end) / 2))
		current := nums[middle]

		if current == target {
			return middle
		}

		if target > current {
			start = middle + 1
		} else {
			end = middle - 1
		}
	}

	return -1
}
